sdv/datasets/demo.py

```diff
@@ -384,6 +384,19 @@ def _get_text_file_content(modality, dataset_name, filename, output_filepath=Non
             The decoded text contents if the file exists, otherwise ``None``.
     """
     _validate_modalities(modality)
+    if output_filepath is not None and not str(output_filepath).endswith('.txt'):
+        fname = (filename or '').lower()
+        if 'readme' in fname:
+            raise ValueError(
+                'The README can only be saved as a txt file. '
+                "Please provide a filepath ending in '.txt'"
+            )
+        if 'source' in fname:
+            raise ValueError(
+                'The source can only be saved as a txt file. '
+                "Please provide a filepath ending in '.txt'"
+            )
```

Comment:
you can simplify this to
```suggestion
        file_type = 'README' if 'readme' in fname else 'source'
        raise ValueError(
            f'The {file_type} can only be saved as a txt file. '
            "Please provide a filepath ending in '.txt'"
        )
```